src/main/kotlin/org/rust/cargo/runconfig/CargoCommandConfiguration.kt

```diff
@@ -25,6 +25,7 @@ class CargoCommandConfiguration(project: Project,
 
     var command: String = "run"
     var additionalArguments: String = ""
+    var environmentVariables: Map<String, String>? = null
```

Comment:
Ah, that's a tough bug!

TLDR: to fix it, just use `var environmentVariables: Map<String, String> = mutableMapOf()`


To understand what is happening here, we need to understand Kotlin collection types. The official docs are here: https://kotlinlang.org/docs/reference/collections.html, what follows is my rough interpretation.

There are mutable collections, read only collections and immutable collections. If you have an immutable collection, you can be sure that it will never change. If you have a read only collection, you can't change it yourself, but somebody else can! A view into mutable collection is a readonly collection. Immutable implies read only, but vice verse is not true.

In Java, there is a single interface for all collection sorts, and readonly/immutable flavors just throw an exception in update operations.

In Kotlin, `MutableCollection` interface corresponds to Java `Collection`. However Kotlin also provides `Collection` interface, which is a restricted readonly version of `MutableCollection`. Kotlin does not have a special interface for truly immutable collections.

So, `Map<String, String>` is a readonly map, which can be backed up by either mutable or immutable map.

But `emptyMap()` is a singleton **immutable** map.


The catch is, `CommandConfiguration` is deserialized from XML, and for collections, IDEAs serializer would just `put` elements into them, so the serializer requires that the underlying collection is mutable.

By using `var environmentVariables: Map<String, String> = mutableMapOf()` we achieve most benefits:

* we use an empty map instead of null (Null Object pattern) 
* we restrict ourselves to readonly portion of the interface
* the serializer is able to read the class from XML, because the underlying map is mutable.